edits to discussion (#164)
SORTEE-Github-Hackathon/manuscript: content/06.discussion-a.md

## Discussion

<!-- ### General paragraph on what GitHub can enable in EcoEvo  
*Contributors to this section: Rob, Brandon*   -->

There have been many calls for researchers outside of the software development community to leverage GitHub for their collaborative research.  
These calls come in light of the continual shift toward science that is more open as well as computationally and data intensive.  
Until now, resources and practical guidance specifically focused on using GitHub within the EEB community have been dispersed in blog posts and video tutorials.  
While these resources have been extremely useful for learning to use GitHub, we provide a more comprehensive review of the key benefits that GitHub can bring to EEB researchers regardless of career stage or coding experience.  

> In this paper, we describe 13 tractable ways that EEB researchers can leverage GitHub to enable more transparent and collaborative research (Figure 2).  

Reviewer: ```suggestion
In this paper, we describe 13 tractable ways that EEB researchers can leverage GitHub to enable more transparent and collaborative research (Figure @fig:scatterblob).  
```
Which figure is this supposed to reference?  If it's `@fig:scatterblob`, then my only hesitation is that there are 15 things on that figure, not 13.

Author: Agreed that this will likely stand out so we should figure out what we want to do. I think that there are 15 in the figure because we have subsections for 13 use cases, but then a couple are also captured under the 'additional uses' section. @LunaSare I think we discussed this a bit in one of our meetings about the figure, but wondering what you all think we should do? Remove the citation here? Elaborate somewhere on why this figure has 15 but we have only 13 sub-headings?

Many of the examples are specifically meant for first-time GitHub users and can likely be adopted with just several hours of practice (e.g., storing data, creating virtual notebooks, making code citable).  
For example, storing code and data and making it citable simply involves creating a repository on Github, pushing code to the repository, and then going through several quick steps to create a DOI for the repository.   
Even with time necessary to learn some GitHub and version control basics, we emphasize that these are transferrable skills that can help students and postdocs be successful in academia or industry.  

Other more advanced examples in this paper may require a greater time commitment, but have the potential to make EEB research even more open, accessible, and collaborative than ever before.  
For example, managing full research projects or research labs on GitHub will require careful thought as to how to delegate tasks such as reviewing pull requests or creating issues.  

> Additionally, collaboratively writing a manuscript using GitHub, as we have done here, involves an entire team to be engaged in more advanced GitHub processes like using GitHub actions and reviewing pull requests.   

Reviewer: ```suggestion
Additionally, collaboratively writing a manuscript using GitHub, as we have done here, involves an entire team to be engaged in more advanced GitHub processes like making and reviewing pull requests.   
```
While GitHub actions was important, the *whole team* didn't need to know it.  It was helpful (essential even) to have someone (@pedrohbraga) on the team to spearhead the setup and maintenance of the more technical aspects of GitHub.

### Limitations to using GitHub for research collaboration
<!-- *Contributors to this section: Rob*, Ali --> 
Though we see GitHub as a useful tool for collaboration in EEB, we describe 2 use cases where, to our knowledge, GitHub's features still fall short of those necessary for highly collaborative EEB research (Table 2).  
First, because of the underlying version control system which tracks "pushed" changes through "commits", real-time collaborative editing (e.g., as on a shared Google Doc or Word document stored on Dropbox) is not possible on GitHub.  
There are now websites that are built on top of the GitHub architecture that allow real-time collaborative editing on a document that can then be pushed to GitHub (e.g., [hackMD](https://hackmd.io/).  
We used HackMD at two key points in writing our manuscript when real-time co-writing was essential: when taking meeting notes and writing the outline of our paper.   
Second, we looked to other software when working on figures and tables.  

> Though creating tables and figures on GitHub using markdown or other scripting languages is possible, we found that it was not practical at the early brainstorming stages.

Reviewer: ```suggestion
Though creating tables and figures on GitHub using markdown or other scripting languages is possible, we found that it was not practical at the early brainstorming stages or for creating publication quality tables and figures.
```
We ended up needing other software to make and format some of the tables and all of the figures.

We needed to rapidly iterate on figure and table design, share feedback through comments, and merge/reorder ideas when necessary.
For these reasons, we used Google Slides for working on figures and Google Sheets for working on tables.
As our figures and tables moved toward more finalized forms, some co-authors chose to create the tables and figures using R and Markdown which could then be tracked using the same version control system as the rest of manuscript.

There are also more logistical constraints to using GitHub in research workflows.  
Due to some file size limitations imposed by GitHub it can be difficult to synch large files to GitHub repos.  
One way around this issue is to attach additional large files to the repository release once code in the repository is ready for use.  
Another limitation is the lack of GitHub help documents for non-English researchers in EEB that may lead them to miss the opportunity to fully understand the importance version control as well as the other benefits of GitHub.  

> Lastly, GitHub is owned by a private company (microsoft) and though the platform hosts an immense amount of open source code, researchers should also consider archiving research materials in long-term data and code repositories.  

Reviewer: ```suggestion
Lastly, GitHub is owned by a private company (Microsoft) and though the platform hosts an immense amount of open source code, researchers should also consider archiving research materials in long-term data and code repositories.  
```

> ### Why aren't more EEB researchers using GitHub?

Reviewer: The edits here are great, but it feels like it belongs better in the introduction.  Fine for now, but let's revisit in our big read-through.

Author: Sounds good, I think that's the type of big picture changes we can make in our next readthrough.

The default public nature of GitHub usage can add additional pressure to students learning to use the platform (Gomes et al. In Prep).   
Thus, hesitancy may come from a more general reluctance to share data and code publicly [@doi:10.1038/s41597-021-00981-0], or technical and logistical issues related to storage of large data files and lack of integration with other research platforms.  

While many quantitative EEB researchers may take advantage of GitHub for individual use, collaborative use may lag behind due to how researchers traditionally divide labor within projects.  
GitHub remains a tool predominantly used by computer scientists and software developers, and EEB researchers may view GitHub as a platform that only needs to be used by individuals writing code.      
Those assumptions may have obscured the utility of GitHub for tasks other than traditional data analysis and software development, or how GitHub can facilitate the integration of code with non-coding aspects of projects through the practice of repository design.  

> ### Box 1

Reviewer: This is Box 2 now, right?

Author: Great, thanks! I just updated that to box 2

<!--*Contributors to this section: Ali, Emma* -->

| 10 Tips for getting started in GitHub |
|---|
| 1. **Check for an existing solution to your problem** The GitHub Help [webpage](https://docs.github.com/en) contains extensive and detailed documents with helpful screenshots. It is a good starting point for handling an issue, and has troubleshooting tips for specific problems. Alternatively, consider Tweeting your issue. There is a large community of GitHub users around the world who have likely faced analogous problems and may be able to provide quick solutions. Third, try to follow blogs e.g.(https://github.blog/), Twitter accounts or YouTube channels that regularly post practical solutions about common GitHub issues. |

> | 2- **Consider taking free courses** such as those from [Software Carpentries](https://swcarpentry.github.io/git-novice/) and sharing these courses with your lab members or colleagues.|

Reviewer 2: this number needs a period for consistency instead of a dash (also in hypothes.is comments)

Author: Made the update, thanks!
